org.adempiere.ui.zk/WEB-INF/src/org/adempiere/webui/window/DateRangeEditor.java

```diff
@@ -0,0 +1,142 @@
+/**********************************************************************
+* This file is part of iDempiere ERP Open Source                      *
+* http://www.idempiere.org                                            *
+*                                                                     *
+* Copyright (C) Contributors                                          *
+*                                                                     *
+* This program is free software; you can redistribute it and/or       *
+* modify it under the terms of the GNU General Public License         *
+* as published by the Free Software Foundation; either version 2      *
+* of the License, or (at your option) any later version.              *
+*                                                                     *
+* This program is distributed in the hope that it will be useful,     *
+* but WITHOUT ANY WARRANTY; without even the implied warranty of      *
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the        *
+* GNU General Public License for more details.                        *
+*                                                                     *
+* You should have received a copy of the GNU General Public License   *
+* along with this program; if not, write to the Free Software         *
+* Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,          *
+* MA 02110-1301, USA.                                                 *
+*                                                                     *
+* Contributors:                                                       *
+* - Peter Takacs, Cloudempiere                                        *
+**********************************************************************/
+package org.adempiere.webui.window;
+
+import org.adempiere.webui.LayoutUtils;
+import org.adempiere.webui.component.Button;
+import org.adempiere.webui.component.Textbox;
+import org.adempiere.webui.component.ToolBarButton;
+import org.adempiere.webui.component.ZkCssHelper;
+import org.adempiere.webui.editor.WEditor;
+import org.adempiere.webui.event.ValueChangeEvent;
+import org.adempiere.webui.event.ValueChangeListener;
+import org.adempiere.webui.theme.ThemeManager;
+import org.zkoss.zk.ui.event.Events;
+import org.zkoss.zul.Div;
+
+/**
+*
+* @author Peter Takacs, Cloudempiere
+*
+*/
+public class DateRangeEditor extends Div implements ValueChangeListener {
+
+	/**
+	 *
+	 */
+	private static final long serialVersionUID = 3329360781729704243L;
+
+	private static final String IMAGES_CONTEXT_CALENDAR_PNG = "images/Calendar16.png";
+
+	private Textbox textbox;
+	ToolBarButton popupBtn;
```

Comment:
most of the time we should use protected instead of package visibility for instance variables.